bitcoin/BRPeer.c

```diff
@@ -1111,6 +1110,22 @@ static void *_peerThreadRoutine(void *arg)
     return NULL; // detached threads don't need to return a value
 }
 
+static void *_peerThreadDisconnectRoutine(void *arg) {
+    BRPeer *peer = arg;
+    BRPeerContext *ctx = arg;
+    
+    pthread_cleanup_push(ctx->threadCleanup, ctx->info);
+
+    peer_log(peer, "waiting-disconnected");
+
+    assert (0 == array_count(ctx->pongCallback));
+    assert (NULL == ctx->mempoolCallback);
```

Comment:
These probably aren't needed. Added them due to my unfamiliarity with this aspect of the BTC code (assert would catch if we needed to handle this).

Reply:
```suggestion
    assert (0 == array_count(ctx->pongCallback));
    assert (NULL == ctx->mempoolCallback);
```